internal/modx/modx.go

```diff
@@ -0,0 +1,26 @@
+package modx
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"golang.org/x/mod/modfile"
+)
+
+// Open the go.mod file in the given directory
+func Open(dir string) (m *modfile.File, err error) {
```

Comment:
this package will centralize mutations to the go module file. making it easier to add my proposed change for file resolution.